File: app/src/main/java/com/squizzard/about/AboutActivity.java
package com.squizzard.about;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.view.View.OnClickListener;

import com.squizzard.analytics.AnalyticsHelper;
import com.squizzard.misriCalendar.R;
import com.squizzard.util.Utility;

import androidx.appcompat.app.AppCompatActivity;

public class AboutActivity extends AppCompatActivity implements OnClickListener{

	final Intent emailIntent = new Intent(android.content.Intent.ACTION_SEND);
	private AnalyticsHelper analyticsHelper;

	protected void onCreate(Bundle savedInstanceState){
		super.onCreate(savedInstanceState);
		analyticsHelper = new AnalyticsHelper(getApplicationContext());
		setContentView(R.layout.about);
		if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }
		findViewById(R.id.emailButton).setOnClickListener(this);
	}

	@Override
	public boolean onOptionsItemSelected(MenuItem menuItem) {
	    onBackPressed();
	    return true;
	}

	public void onClick(View v) {
		if (v.getId() == R.id.emailButton){
			analyticsHelper.sendEvent("contact_developer");

			String messageString = "\n\n\n\n OS Version: " + System.getProperty("os.version");
			messageString += "\n OS API Level: " + Build.VERSION.RELEASE;
			messageString += "\n Device: " + android.os.Build.DEVICE;
			messageString += "\n Model : " + android.os.Build.MODEL;
			messageString += "\n Display: " + android.os.Build.DISPLAY;
			messageString += "\n Manufacturer: " + android.os.Build.MANUFACTURER;
			emailIntent.setType("plain/text");
			emailIntent.putExtra(android.content.Intent.EXTRA_EMAIL, new String[]{getString(R.string.ouikka_email)});
			emailIntent.putExtra(android.content.Intent.EXTRA_SUBJECT,
                    "MisriCal "+ Utility.getVersion(getApplicationContext()));
			emailIntent.putExtra(android.content.Intent.EXTRA_TEXT, messageString);
			emailIntent.setType("message/rfc822");
			startActivity(Intent.createChooser(emailIntent, "Send mail..."));
		}	
	}
}

File: app/src/main/java/com/squizzard/about/AboutActivity.kt
package com.squizzard.about

import com.squizzard.analytics.AnalyticsHelper.sendEvent
import androidx.appcompat.app.AppCompatActivity
import android.content.Intent
import android.os.Bundle
import com.squizzard.misriCalendar.R
import android.os.Build
import android.view.MenuItem
import android.view.View
import com.squizzard.misriCalendar.BuildConfig

class AboutActivity : AppCompatActivity(), View.OnClickListener {
    private val emailIntent = Intent(Intent.ACTION_SEND)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.about)
        if (supportActionBar != null) {
            supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        }
        findViewById<View>(R.id.emailButton).setOnClickListener(this)
    }

    override fun onOptionsItemSelected(menuItem: MenuItem): Boolean {
        onBackPressedDispatcher.onBackPressed()
        return true
    }

    override fun onClick(v: View) {
        if (v.id == R.id.emailButton) {
            sendEvent("contact_developer")
            var messageString = """
            OS Version: ${System.getProperty("os.version")}"""
            messageString += """
            OS API Level: ${Build.VERSION.RELEASE}"""
            messageString += """
            Device: ${Build.DEVICE}"""
            messageString += """
            Model : ${Build.MODEL}"""
            messageString += """
            Display: ${Build.DISPLAY}"""
            messageString += """
            Manufacturer: ${Build.MANUFACTURER}"""
            emailIntent.type = "plain/text"
            emailIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.ouikka_email)))
            emailIntent.putExtra(
                Intent.EXTRA_SUBJECT,
                "MisriCal " + BuildConfig.VERSION_NAME
            )
            emailIntent.putExtra(Intent.EXTRA_TEXT, messageString)
            emailIntent.type = "message/rfc822"
            startActivity(Intent.createChooser(emailIntent, "Send mail..."))
        }
    }
}